src/main/java/be/acara/events/service/EventService.java

```diff
@@ -54,7 +56,25 @@ private Event getEvent(long id) {
     }
 
     public EventDto addEvent(EventDto eventDto) {
+        if (eventDto.getId() != null) {
+            throw new IdAlreadyExistsException("A new entity cannot already contain an id");
+        }
         Event event = mapper.map(eventDto);
         return mapper.map(repository.saveAndFlush(event));
     }
+
+    public EventDto editEvent(long id, EventDto eventDto) {
+        Event eventToEdit = getEvent(id);
+        if(eventDto.getId().equals(eventToEdit.getId())){
+            Event event = mapper.map(eventDto);
+            return mapper.map(repository.saveAndFlush(event));
+        }
+        else {
+            throw new BadRequestException(
+                    "Bad request on editing member",
+                    String.format(
+                            "Id of member to edit does not match given id. Member id = %d, and given id = %d", eventDto.getId(), id)
```

Comment:
After examination, it appears my choice of 422 Unprocessable Entity was wrong. However, you should still try to use the IdAlreadyExistsException and instead make the exception return 400 instead of 422.